File: tests/python/testC.py
class TestC:
    @staticmethod
    def print_hello_test_c1():
        print("Hello, World!")
        for i in range(5):
            print(f"This is line {i}")
            if i % 2 == 0:
                print("Even number")
            else:
                print("Odd number")
        print("End of printHello")

    @staticmethod
    def print_hello_test_c2():
        print("Hello, World!")
        for i in range(5):
            print(f"This is line {i}")
            if i % 2 == 0:
                print("Even number")
            else:
                print("Odd number")
            print(f"Current iteration: {i}")
        print("End of printHelloAgain")

    @staticmethod
    def main():
        TestC.print_hello_test_c1()
        TestC.print_hello_test_c2()

# Execute main method
TestC.main()

File: tests/typescript/testC.ts
class TestC {
    static printHelloTestC1(): void {
        console.log("Hello, World!");
        for (let i: number = 0; i < 5; i++) {
            console.log(`This is line ${i}`);
            if (i % 2 === 0) {
                console.log("Even number");
            } else {
                console.log("Odd number");
            }
        }
        console.log("End of printHello");
    }

    static printHelloTestC2(): void {
        console.log("Hello, World!");
        for (let i: number = 0; i < 5; i++) {
            console.log(`This is line ${i}`);
            if (i % 2 === 0) {
                console.log("Even number");
            } else {
                console.log("Odd number");
            }
            console.log(`Current iteration: ${i}`);
        }
        console.log("End of printHelloAgain");
    }

    static main(): void {
        this.printHelloTestC1();
        this.printHelloTestC2();
    }
}

// Execute main method
TestC.main();